### PR TITLE
feat: api for deleting a review

### PR DESCRIPTION
**Pull request by chinathaip**
- endpoint that allows users to delete their review
   - protected by JWT
   - only review owner can delete

- migrate `review_repo_impl_integration_test.go` from "testify/assert" to "testify/suite" for better test lifecycle handling (setting up and tearing down instances)

- allow multiple path parameters in api gateway (idk why I didn't do it in the first place)

- send rejected reason to frontend if any

### CODE REVIEW

File: apps/api-gateway/gateway-dev.yaml [hunk 1]
```diff
@@ -19,8 +19,14 @@ routes:
...
+    - path: /api/courses/:courseCode/reviews
       method: POST
       destination:
           service: submit_new_review
           endpoint: http://localhost:8003/api/courses/:courseCode/reviews
+
+    - path: /api/courses/:courseCode/reviews/:reviewId
+      method: DELETE
+      destination:
+          service: submit_new_review
```
Comment (Lxkas) [inline apps/api-gateway/gateway-dev.yaml:31]: method delete, service name submit_new XD

File: apps/course-api/package.json [hunk 2]
```diff
@@ -11,7 +11,8 @@
     "integration": "docker compose -f docker-compose.integration.yaml down -v && docker compose -f docker-compose.integration.yaml up --build --force-recreate --abort-on-container-exit --exit-code-from app",
     "integration:jest": "jest test/integration --detectOpenHandles",
     "docker:test_build": "docker compose -f docker-compose.test-build.yaml down -v && docker compose -f docker-compose.test-build.yaml up -d --build --force-recreate",
-    "prisma:generate_types": "pnpm prisma generate --schema=../../packages/prisma/schema.prisma"
+    "bootstrap:unix": "pnpm prisma generate --schema=../../packages/prisma/schema.prisma",
+    "bootstrap:windows": "pnpm prisma generate --schema=../../packages/prisma/schema.prisma"
```
Comment (Lxkas) [inline apps/course-api/package.json:15]: ??

File: apps/api-gateway/gateway-dev.yaml (hunk 1 above)
Comment (chinathaip) [reply apps/api-gateway/gateway-dev.yaml:31]: good catch!

File: apps/course-api/package.json (hunk 2 above)
Comment (chinathaip) [reply apps/course-api/package.json:15]: it's an attempt to generate prisma for course api automatically when running either `pnpm run bootstrap:unix` or `pnpm run bootstrap:windows`. (The commands must be the same for turbo to detect, unless there's a shortcut for this XD)

**Pull request merged**